orca-core/src/main/groovy/com/netflix/spinnaker/orca/batch/exceptions/ExceptionHandler.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.batch.exceptions;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * ExceptionHandler.
+ */
+public interface ExceptionHandler {
+  boolean handles(Exception e);
+
+  Response handle(String taskName, Exception exception);
+
+  class Response {
+    private final Date timestamp = new Date();
+    private String exceptionType;
+    private String operation;
+    private ResponseDetails details;
+    private boolean shouldRetry = false;
+
+    public Response() {
+
+    }
+
+    public Response(String exceptionType, String operation, ResponseDetails responseDetails, boolean shouldRetry) {
+      this.exceptionType = exceptionType;
+      this.operation = operation;
+      this.details = responseDetails;
+      this.shouldRetry = shouldRetry;
+    }
+
+    public Date getTimestamp() {
+      return timestamp;
+    }
+
+    public String getExceptionType() {
+      return exceptionType;
+    }
+
+    public void setExceptionType(String exceptionType) {
```

Comment:
it was `@Canonical` in the groovy world and I didn't want to necessarily chase down all usages (though I actually probably should just do that and make this immutable)